src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -186,9 +195,10 @@ function ShallowWaterModel(;
     closure = with_tracers(tracernames(tracers), closure)
 
     model = ShallowWaterModel(grid,
+                              vertical_model,
                               arch,
                               clock,
-                              eltype(grid)(gravitational_acceleration),
+                              gravitational_acceleration,
```

Comment:
if `gravitational_acceleration` is a number then we better ensure that it has the same type as the grid to avoid type instability

```suggestion
                              eltype(grid)(gravitational_acceleration),
```

Reply:
Yes, I agree with you. I changed it because as of now, I wrote the code assuming a single `gravitational_acceleration` variable instead of two.